src/blob/index/core.rs

```diff
@@ -81,12 +84,15 @@ impl<FileIndex: FileIndexTrait> IndexStruct<FileIndex> {
     pub(crate) fn clear(&mut self) {
         self.inner = State::InMemory(BTreeMap::new());
         self.mem = Some(Default::default());
-        self.filter.clear();
+        self.bloom_filter.clear();
+        self.range_filter.clear();
     }
 
-    pub fn check_bloom_key(&self, key: &[u8]) -> Option<bool> {
-        if self.params.filter_is_on {
-            Some(self.filter.contains(key))
+    pub(crate) fn check_filters_key(&self, key: &[u8]) -> Option<bool> {
+        if !self.range_filter.contains(key) {
```

Comment:
There is an error in this condition chain. If `range_filter` returns `true` and bloom filter is off, the result should be `true`, not `None`

Reply:
Actually, there is no, but I rewrite this function a bit and now it returns just bool.
Previous version returned:
1) None (if key can't be checked due to bloom filter off state) aka 'no information, go and check'
2) Some(true) (if key exists according to bloom filter (it means "go and check explicitly")) aka 'likely exists but check'
3) Some(false) (if key doesn't exist **for sure**) aka 'not for sure'

Now I decided to unite None and Some(true) variants and return just false (doesn't exist for sure) and true (maybe exists)

This code is okay and here's it's logic:
1. if range filters **doesn't** contains this key (aka 'is key < min or key > max?') => return Some(false) aka 'not for sure'
2. if bloom is on then return Some(bloom_result), where bloom_result either true (=> then return Some(true) aka 'likely exists but check') or false (=> then return Some(false) aka 'not for sure')
3. if bloom is off then return None aka 'no information, go and check' (Some(true) aka 'likely exists but check' also can be returned in this case, but it doesn't matter)

So to not confuse someone else I made it return just true or false.